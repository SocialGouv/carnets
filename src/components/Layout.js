import React from "react";
import Head from "next/head";

import Nav from "./Nav";

import { UserProvider } from "../lib/user";

const CI_COMMIT_BEFORE_SHA = process.env.CI_COMMIT_BEFORE_SHA;
const CI_COMMIT_REF_SLUG = process.env.CI_COMMIT_REF_SLUG;
const CI_COMMIT_REF_NAME = process.env.CI_COMMIT_REF_NAME;
const CI_COMMIT_SHORT_SHA = process.env.CI_COMMIT_SHORT_SHA;
const CI_COMMIT_SHA = process.env.CI_COMMIT_SHA;

const Layout = ({ user, loading = false, children }) => (
  <UserProvider value={{ user, loading }}>
    <Head>
      <title>@SocialGouv/carnets</title>
    </Head>

    <Nav />

    <div className="container" style={{ paddingTop: 40 }}>
      {children}
    </div>
    <div>
      <div>CI_COMMIT_BEFORE_SHA: {CI_COMMIT_BEFORE_SHA}</div>
      <div>CI_COMMIT_REF_SLUG: {CI_COMMIT_REF_SLUG}</div>
      <div>CI_COMMIT_REF_NAME: {CI_COMMIT_REF_NAME}</div>
      <div>CI_COMMIT_SHORT_SHA: {CI_COMMIT_SHORT_SHA}</div>
      <div>CI_COMMIT_SHA: {CI_COMMIT_SHA}</div>
    </div>
  </UserProvider>
);

export default Layout;
