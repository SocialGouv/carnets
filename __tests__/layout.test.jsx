import { vi, expect, test } from "vitest";
import { render } from "@testing-library/react";
import Layout from "../src/app/layout";
import Page from "../src/app/page";

vi.mock("next-auth/next", async (importOriginal) => {
  const actual = await importOriginal();
  return {
    ...actual,
    getServerSession: () => {
      user: "bob";
    },
  };
});

test("Page", async () => {
  const { container } = render(await Layout({ children: Page }));
  expect(container).toMatchSnapshot();
});
