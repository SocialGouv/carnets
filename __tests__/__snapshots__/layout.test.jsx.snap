// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page 1`] = `
<div>
  <html
    lang="fr"
  >
    <head>
      <link
        as="font"
        crossorigin="anonymous"
        href="/node_modules/@codegouvfr/react-dsfr/dsfr/fonts/Marianne-Regular.woff2"
        rel="preload"
      />
      <link
        as="font"
        crossorigin="anonymous"
        href="/node_modules/@codegouvfr/react-dsfr/dsfr/fonts/Marianne-Medium.woff2"
        rel="preload"
      />
      <link
        as="font"
        crossorigin="anonymous"
        href="/node_modules/@codegouvfr/react-dsfr/dsfr/fonts/Marianne-Bold.woff2"
        rel="preload"
      />
      <link
        href="/node_modules/@codegouvfr/react-dsfr/dsfr/favicon/apple-touch-icon.png"
        rel="apple-touch-icon"
      />
      <link
        href="/node_modules/@codegouvfr/react-dsfr/dsfr/favicon/favicon.svg"
        rel="icon"
        type="image/svg+xml"
      />
      <link
        href="/node_modules/@codegouvfr/react-dsfr/dsfr/favicon/favicon.ico"
        rel="shortcut icon"
        type="image/x-icon"
      />
      <script>
        
{

    window.ssrWasPerformedWithIsDark = "system" === "dark";
	const sanitizer = typeof trustedTypes !== "undefined" ? trustedTypes.createPolicy("react-dsfr-asap", { createHTML: s =&gt; s }) : {
		createHTML: s =&gt; s,
	};
    
    const isDark = (() =&gt; {
    
    	const isDarkExplicitlyProvidedAsParameter = (() =&gt; {
    		if ("system" === "system") {
    			return undefined;
    		}
    
    		switch ("system") {
    			case "dark": return true;
    			case "light": return false;
    		}
    	})();
    
    	const isDarkFromLocalStorage = (() =&gt; {
    		const colorSchemeReadFromLocalStorage = localStorage.getItem("scheme");
    
    		if (colorSchemeReadFromLocalStorage === null) {
    			return undefined;
    		}
    
    		if (colorSchemeReadFromLocalStorage === "system") {
    			return undefined;
    		}
    
    		switch (colorSchemeReadFromLocalStorage) {
    			case "dark":
    				return true;
    			case "light":
    				return false;
    		}
    	})();
    
    	const isDarkFromOsPreference = (() =&gt; {
    		if (!window.matchMedia) {
    			return undefined;
    		}
    
    		return window.matchMedia("(prefers-color-scheme: dark)").matches;
    	})();
    
    	const isDarkFallback = false;
    
    	return (
    		isDarkFromLocalStorage ??
    		isDarkExplicitlyProvidedAsParameter ??
    		isDarkFromOsPreference ??
    		isDarkFallback
    	);
    
    })();
    
    ["data-fr-scheme", "data-fr-theme"].forEach(attr =&gt; document.documentElement.setAttribute(attr, isDark ? "dark" : "light"));

    {

        let element= document.getElementById("dsfr-root-color-scheme");

		if( element !== null ){
			element.remove()
		}

        element = document.createElement("style");

		if ("" !== "") {
			element.setAttribute("nonce", "");
		}

        element.id = "dsfr-root-color-scheme";

        element.innerHTML = sanitizer.createHTML(\`:root { color-scheme: \${isDark ? "dark" : "light"}; }\`);

        document.head.appendChild(element);

    }

    {

        const name = "theme-color";

        let element = document.querySelector(\`meta[name=\${name}]\`);
		
		if( element !== null ){
			element.remove();
		}

        element = document.createElement("meta");

        element.name = name;

        element.content = isDark ? "#161616" : "#ffffff";

        document.head.appendChild(element);

    }

}

      </script>
    </head>
    <body>
      <button
        aria-controls="fr-theme-modal"
        aria-hidden="true"
        class="fr-btn fr-hidden"
        data-fr-js-modal-button="true"
        data-fr-opened="false"
        id="fr-theme-modal-hidden-control-button"
        tabindex="-1"
        type="button"
      >
         
      </button>
      <dialog
        aria-labelledby="fr-modal-title-fr-theme-modal"
        class="fr-modal"
        data-fr-concealing-backdrop="true"
        data-fr-js-modal="true"
        id="fr-theme-modal"
        role="dialog"
      >
        <div
          class="fr-container fr-container--fluid fr-container-md"
        >
          <div
            class="fr-grid-row fr-grid-row--center"
          >
            <div
              class="fr-col-12 fr-col-md-6 fr-col-lg-4"
            >
              <div
                class="fr-modal__body"
                data-fr-js-modal-body="true"
              >
                <div
                  class="fr-modal__header"
                >
                  <button
                    aria-controls="fr-theme-modal"
                    class="fr-btn--close fr-btn"
                    data-fr-js-modal-button="true"
                    title="Fermer"
                    type="button"
                  >
                    Fermer
                  </button>
                </div>
                <div
                  class="fr-modal__content"
                >
                  <h1
                    class="fr-modal__title"
                    id="fr-modal-title-fr-theme-modal"
                  >
                    Paramètres d'affichage
                  </h1>
                  <div
                    class="fr-display"
                    data-fr-js-display="true"
                  >
                    <div
                      class="fr-form-group"
                    >
                      <fieldset
                        class="fr-fieldset"
                      >
                        <legend
                          class="fr-fieldset__legend fr-text--regular"
                        >
                          Choisissez un thème pour personnaliser l'apparence du site.
                        </legend>
                        <div
                          class="fr-fieldset__content"
                        >
                          <div
                            class="fr-radio-group fr-radio-rich"
                          >
                            <input
                              id=":r0:-light"
                              name="fr-radios-theme"
                              type="radio"
                              value="light"
                            />
                            <label
                              class="fr-label"
                              for=":r0:-light"
                            >
                              Thème clair
                            </label>
                            <div
                              class="fr-radio-rich__img"
                            >
                              <svg
                                aria-hidden="true"
                                height="80px"
                                viewBox="0 0 80 80"
                                width="80px"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <use
                                  class="fr-artwork-decorative"
                                  xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/light.svg#artwork-decorative"
                                />
                                <use
                                  class="fr-artwork-minor"
                                  xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/light.svg#artwork-minor"
                                />
                                <use
                                  class="fr-artwork-major"
                                  xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/light.svg#artwork-major"
                                />
                              </svg>
                            </div>
                          </div>
                          <div
                            class="fr-radio-group fr-radio-rich"
                          >
                            <input
                              id=":r0:-dark"
                              name="fr-radios-theme"
                              type="radio"
                              value="dark"
                            />
                            <label
                              class="fr-label"
                              for=":r0:-dark"
                            >
                              Thème sombre
                            </label>
                            <div
                              class="fr-radio-rich__img"
                            >
                              <svg
                                aria-hidden="true"
                                height="80px"
                                viewBox="0 0 80 80"
                                width="80px"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <use
                                  class="fr-artwork-decorative"
                                  xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/dark.svg#artwork-decorative"
                                />
                                <use
                                  class="fr-artwork-minor"
                                  xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/dark.svg#artwork-minor"
                                />
                                <use
                                  class="fr-artwork-major"
                                  xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/dark.svg#artwork-major"
                                />
                              </svg>
                            </div>
                          </div>
                          <div
                            class="fr-radio-group fr-radio-rich"
                          >
                            <input
                              id=":r0:-system"
                              name="fr-radios-theme"
                              type="radio"
                              value="system"
                            />
                            <label
                              class="fr-label"
                              for=":r0:-system"
                            >
                              Système.
                              <span
                                class="fr-hint-text"
                              >
                                Utilise les paramètres système.
                              </span>
                            </label>
                            <div
                              class="fr-radio-rich__img"
                            >
                              <svg
                                aria-hidden="true"
                                height="80px"
                                viewBox="0 0 80 80"
                                width="80px"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <use
                                  class="fr-artwork-decorative"
                                  xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/system.svg#artwork-decorative"
                                />
                                <use
                                  class="fr-artwork-minor"
                                  xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/system.svg#artwork-minor"
                                />
                                <use
                                  class="fr-artwork-major"
                                  xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/system.svg#artwork-major"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </dialog>
      <header
        class="fr-header"
        id="fr-header-simple-header"
        role="banner"
      >
        <div
          class="fr-header__body"
        >
          <div
            class="fr-container"
          >
            <div
              class="fr-header__body-row"
            >
              <div
                class="fr-header__brand fr-enlarge-link"
              >
                <div
                  class="fr-header__brand-top"
                >
                  <div
                    class="fr-header__logo"
                  >
                    <p
                      class="fr-logo"
                    >
                      Ministères
                      <br />
                      Sociaux
                    </p>
                  </div>
                  <div
                    class="fr-header__operator"
                  >
                    <a
                      href="/"
                      title="Accueil - La Fabrique Numérique des ministères sociaux"
                    >
                      <img
                        alt="logo de la Fabrique Numérique"
                        class="fr-responsive-img"
                        data-fr-js-ratio="true"
                        src="/logo-fabrique.svg"
                        style="width: 3.5rem; height: 0px;"
                      />
                    </a>
                  </div>
                  <div
                    class="fr-header__navbar"
                  >
                    <button
                      aria-controls="header-menu-modal-fr-header-simple-header"
                      aria-haspopup="menu"
                      class="fr-btn--menu fr-btn"
                      data-fr-js-modal-button="true"
                      data-fr-opened="false"
                      id="fr-header-simple-header-menu-button"
                      title="Menu"
                    >
                      Menu
                    </button>
                  </div>
                </div>
                <div
                  class="fr-header__service"
                >
                  <a
                    href="/"
                    title="Accueil - La Fabrique Numérique des ministères sociaux"
                  >
                    <p
                      class="fr-header__service-title"
                    >
                      Carnets
                    </p>
                  </a>
                  <p
                    class="fr-header__service-tagline"
                  >
                    Carnets de bord des produits de la Fabrique Numérique
                  </p>
                </div>
              </div>
              <div
                class="fr-header__tools"
              >
                <div
                  class="fr-header__tools-links"
                  data-fr-js-header-links="true"
                >
                  <ul
                    class="fr-btns-group"
                  >
                    <li>
                      <a
                        class="fr-btn fr-icon-line-chart-line"
                        href="/stats"
                        id="fr-header-simple-header-quick-access-item--Statistiques-0"
                      >
                        Statistiques
                      </a>
                    </li>
                    <li>
                      <button
                        class="fr-btn fr-icon-add-circle-line"
                        id="fr-header-simple-header-quick-access-item--Connexion-1"
                      >
                        Connexion
                      </button>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-labelledby="fr-header-simple-header-menu-button"
          class="fr-header__menu fr-modal"
          data-fr-js-header-modal="true"
          data-fr-js-modal="true"
          id="header-menu-modal-fr-header-simple-header"
        >
          <div
            class="fr-container"
          >
            <button
              aria-controls="header-menu-modal-fr-header-simple-header"
              class="fr-btn--close fr-btn"
              data-fr-js-modal-button="true"
              id="fr-header-simple-header-mobile-overlay-button-close"
              title="Fermer"
            >
              Fermer
            </button>
            <div
              class="fr-header__menu-links"
            >
              <ul
                class="fr-btns-group"
              >
                <li>
                  <a
                    class="fr-btn fr-icon-line-chart-line"
                    href="/stats"
                    id="fr-header-simple-header-quick-access-item--Statistiques_mobile-0"
                  >
                    Statistiques
                  </a>
                </li>
                <li>
                  <button
                    class="fr-btn fr-icon-add-circle-line"
                    id="fr-header-simple-header-quick-access-item--Connexion_mobile-1"
                  >
                    Connexion
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </header>
      <div
        class="fr-container flex"
      />
      <footer
        class="fr-footer"
        id="fr-footer"
        role="contentinfo"
      >
        <div
          class="fr-container"
        >
          <div
            class="fr-footer__body"
          >
            <div
              class="fr-footer__brand fr-enlarge-link"
            >
              <a
                href="/"
                title="Accueil - La Fabrique Numérique des ministères sociaux"
              >
                <p
                  class="fr-logo"
                >
                  Ministères
                  <br />
                  Sociaux
                </p>
              </a>
            </div>
            <div
              class="fr-footer__content"
            >
              <p
                class="fr-footer__content-desc"
              >
                Carnets est une application permettant de publier et de mettre à jour les supports de présentation du standup de la Fabrique Numérique.
              </p>
              <ul
                class="fr-footer__content-list"
              >
                <li
                  class="fr-footer__content-item"
                >
                  <a
                    class="fr-footer__content-link"
                    href="https://legifrance.gouv.fr"
                    target="_blank"
                    title="legifrance.gouv.fr - ouvre une nouvelle fenêtre"
                  >
                    legifrance.gouv.fr
                  </a>
                </li>
                <li
                  class="fr-footer__content-item"
                >
                  <a
                    class="fr-footer__content-link"
                    href="https://gouvernement.fr"
                    target="_blank"
                    title="gouvernement.fr - ouvre une nouvelle fenêtre"
                  >
                    gouvernement.fr
                  </a>
                </li>
                <li
                  class="fr-footer__content-item"
                >
                  <a
                    class="fr-footer__content-link"
                    href="https://service-public.fr"
                    target="_blank"
                    title="service-public.fr - ouvre une nouvelle fenêtre"
                  >
                    service-public.fr
                  </a>
                </li>
                <li
                  class="fr-footer__content-item"
                >
                  <a
                    class="fr-footer__content-link"
                    href="https://data.gouv.fr"
                    target="_blank"
                    title="data.gouv.fr - ouvre une nouvelle fenêtre"
                  >
                    data.gouv.fr
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="fr-footer__bottom"
          >
            <ul
              class="fr-footer__bottom-list"
            >
              <li
                class="fr-footer__bottom-item"
              >
                <span
                  class="fr-footer__bottom-link"
                >
                  Accessibilité: partiellement conforme
                </span>
              </li>
              <li
                class="fr-footer__bottom-item"
              >
                <button
                  aria-controls="fr-theme-modal"
                  class="fr-footer__bottom-link fr-icon-theme-fill fr-link--icon-left"
                  data-fr-js-modal-button="true"
                  data-fr-opened="false"
                  id="fr-theme-modal-control-button"
                >
                  Paramètres d'affichage
                </button>
              </li>
            </ul>
            <div
              class="fr-footer__bottom-copy"
            >
              <p>
                Sauf mention explicite de propriété intellectuelle détenue par des tiers, les contenus de ce site sont proposés sous
                 
                <a
                  href="https://github.com/etalab/licence-ouverte/blob/master/LO.md"
                  target="_blank"
                  title="licence etalab-2.0 - ouvre une nouvelle fenêtre"
                >
                  licence etalab-2.0
                </a>
              </p>
            </div>
          </div>
        </div>
      </footer>
    </body>
  </html>
</div>
`;
